src/define.ts

```diff
@@ -52,7 +52,7 @@ function define<Result>(config: Config<Result>): Factory<Result> {
 
   // Define a property to differentiate this function during the evaluation
   // phase when the factory is later invoked.
-  factory.__cooky_cutter = "factory" as typeof FACTORY_FUNCTION_KEY;
+  factory.__cooky_cutter = FACTORY_FUNCTION_KEY as typeof FACTORY_FUNCTION_KEY;
```

Comment:
Is this cast still necessary with this change? Curious if we could make this line:

```suggestion
  factory.__cooky_cutter = FACTORY_FUNCTION_KEY;
```

Reply:
This cast is still necessary.
If this cast doesn't exist , `factory.__cooky_cutter`'s type is not `'factory'` but `string`. Thus, error occurs by type check.